browser/ui/sidebar/BUILD.gn

```diff
@@ -8,6 +8,20 @@ import("//extensions/buildflags/buildflags.gni")
 
 assert(toolkit_views)
 
+source_set("sidebar") {
+  sources = [
+    "sidebar_controller.h",
```

Comment:
why are you only moving the header files? targets in ui can have allow_circular_includes_from for now, but as discussed in DM there should be a follow-up to fix sidebar_controller.cc